instrumentation/opentelemetry-instrumentation-pyramid/src/opentelemetry/instrumentation/pyramid/callbacks.py

```diff
@@ -158,10 +166,16 @@ def trace_tween(request):
                     "PyramidInstrumentor().instrument_config(config) is called"
                 )
             elif enabled:
+                if response:
+                    if hasattr(response, "status"):
+                        status = response.status
+                    if hasattr(response, "headerList"):
+                        headerList = response.headerList
+
                 otel_wsgi.add_response_attributes(
                     span,
-                    response_or_exception.status,
-                    response_or_exception.headerlist,
+                    status,
+                    headerList,
```

Comment:
Can these `status` and `headerList` ever be `None`? I would be defensive and make sure to call `add_response_attributes` only when status is set to non null value.

Reply:
Agreed status should be guarded. I believe `headerList` can feasibly be `None`, but as of right now `add_response_attributes` does not actually handle the header list.

I'm updating things to be a bit more defensive overall.